### PR TITLE
Make madvice of ChunkCache configurable

### PR DESCRIPTION
**Pull request by bigsheeper**
/kind improvement

### CODE REVIEW

**Review by yah01 — COMMENTED** [review]
please also apply this to column type in `Column.h`

File: configs/milvus.yaml [hunk 1]
```diff
@@ -268,9 +268,11 @@ queryNode:
   loadMemoryUsageFactor: 1 # The multiply factor of calculating the memory usage while loading segments
   enableDisk: false # enable querynode load disk index, and search on disk index
   maxDiskUsagePercentage: 95
-  cache:
+  cache: # deprecated, TODO: remove it
     enabled: true
     memoryLimit: 2147483648 # 2 GB, 2 * 1024 *1024 *1024
+  chunkCache:
+    madvice: MADV_WILLNEED # The read ahead policy of chunk cache, options: MADV_NORMAL, MADV_RANDOM, MADV_SEQUENTIAL, MADV_WILLNEED, MADV_DONTNEED
```
Comment (yah01) [inline configs/milvus.yaml:275]: 1. I'd like to move this into cache, the deprecated cache config items could be removed, but keep cache itself
2. Rename this item to 'policy', and the value to 'willneed', we need to decouple the policy and the mmap advice, user may have no experience about mmap, and Milvus may add buffer pool

File: internal/core/src/storage/ChunkCache.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ ChunkCache::Prefetch(const std::string& filepath) {
     auto ok =
         madvise(reinterpret_cast<void*>(const_cast<char*>(column->Data())),
                 column->ByteSize(),
-                MADV_WILLNEED);  // TODO: sheep, make it configurable
+                MADV_);
```
Comment (yah01) [inline internal/core/src/storage/ChunkCache.cpp:58]: Also this field better to be renamed

File: internal/core/src/storage/ChunkCache.h [hunk 3]
```diff
@@ -21,11 +21,23 @@
 
 namespace milvus::storage {
 
+extern std::map<std::string, int> Madvice_Map;
```
Comment (yah01) [inline internal/core/src/storage/ChunkCache.h:24]: Same

File: internal/core/src/storage/ChunkCache.h [hunk 4]
```diff
@@ -21,11 +21,23 @@
...
+
 class ChunkCache {
  public:
-    explicit ChunkCache(const std::string& path, ChunkManagerPtr cm)
+    explicit ChunkCache(const std::string& path,
+                        const std::string& madvice,
```
Comment (yah01) [inline internal/core/src/storage/ChunkCache.h:29]: a more efficient construtor:
```c++
ChunkCache(std::string path,
           const std::string& madvice,
           ChunkManagerPtr cm) : path_prefix_(std::move(path)), ...
```

File: internal/core/src/storage/ChunkCacheSingleton.h [hunk 5]
```diff
@@ -39,11 +39,11 @@ class ChunkCacheSingleton {
     }
 
     void
-    Init(const std::string& root_path) {
+    Init(const std::string& root_path, const std::string& madvice) {
```
Comment (yah01) [inline internal/core/src/storage/ChunkCacheSingleton.h:42]: like the constructor, use value type (not reference) and then move them would be more efficient

File: configs/milvus.yaml (hunk 1 above)
Comment (bigsheeper) [reply configs/milvus.yaml:275]: policy may be confused, how about `readAheadPolicy`

**Pull request merged**